tests/test_irods.py

```diff
@@ -1448,3 +1450,99 @@ def test_collection_contents(self, special_paths):
         coll = Collection(special_paths)
         observed = [str(x) for x in coll.contents(recurse=True)]
         assert observed == expected
+
+
+@m.describe("JSON serialization")
+class TestJSON:
+    @m.it("Can serialize a collection to JSON")
+    def test_collection_json_serialize(self, simple_collection):
+        coll = Collection(simple_collection)
+        coll.add_metadata(AVU("a", 1), AVU("b", 2), AVU("c", 3))
+
+        kwargs = {"indent": None, "sort_keys": True}
+
+        assert coll.to_json(**kwargs) == json.dumps(
+            {
+                Baton.COLL: coll.path.as_posix(),
+                Baton.AVUS: [
+                    {Baton.ATTRIBUTE: "a", Baton.VALUE: "1"},
+                    {Baton.ATTRIBUTE: "b", Baton.VALUE: "2"},
+                    {Baton.ATTRIBUTE: "c", Baton.VALUE: "3"},
+                ],
+                Baton.ACCESS: [
+                    {Baton.OWNER: "irods", Baton.ZONE: "testZone", Baton.LEVEL: "own"}
+                ],
+            },
+            **kwargs,
+        )
+
+    def test_collection_json_deserialize(self, simple_collection):
+        coll1 = Collection(simple_collection)
+        coll1.add_metadata(AVU("a", 1), AVU("b", 2), AVU("c", 3))
+
+        json_str = coll1.to_json(indent=None, sort_keys=True)
+
+        coll2 = Collection.from_json(json_str)
+        assert coll2.path == coll1.path
+        assert coll2.metadata() == coll1.metadata()
+        assert coll2.acl() == coll1.acl()
+
+    @m.it("Can serialize a data object to JSON")
+    def test_data_object_json_serialize(self, simple_data_object):
```

Comment:
Perhaps add a test for serializing with additional acls, as well as deserializing with them?